vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleOverviewManager.java

```diff
@@ -0,0 +1,89 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.exception.ModuleNotFoundException;
+import edu.asu.diging.vspace.core.model.IChoice;
+import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.ISequence;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.BranchingPoint;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.IModuleOverviewManager;
+import edu.asu.diging.vspace.core.services.impl.model.ModuleOverview;
+import edu.asu.diging.vspace.core.services.impl.model.SequenceOverview;
+import edu.asu.diging.vspace.core.services.impl.model.SlideOverview;
+
+@Service
+public class ModuleOverviewManager implements IModuleOverviewManager {
+    
+    @Autowired
+    private IModuleManager moduleManager;
+    
+    public ModuleOverview getModuleOverview(String moduleId) throws ModuleNotFoundException{
+        IModule module = moduleManager.getModule(moduleId);
+        if(module==null) {
+            throw new ModuleNotFoundException("Module not found");
+        }
+        ISequence startSequence = module.getStartSequence();
+        
+        List<ISequence> sequences = moduleManager.getModuleSequences(moduleId);
+        
+        SequenceOverview sequenceOverviewNode = createSequenceOverviewNode(startSequence);
+        List<SequenceOverview> otherSequences = new ArrayList<SequenceOverview>();
+        for(ISequence sequence : sequences) {
+            
+            if(sequence != startSequence) {
+                otherSequences.add(createSequenceOverviewNode(sequence));
+            }
+        }
+        
+        ModuleOverview moduleOverview = new ModuleOverview();
+        moduleOverview.setStartSequence(sequenceOverviewNode);
+        moduleOverview.setOtherSequences(otherSequences);
+        return moduleOverview;
+    }
+    
+    /**
+     * This method is used to fetch all Sequences which belong to a module and 
+     * convert this into a SequenceOverview node. The SequenceOverviewNode is added to 
+     * ModuleOverview
+     * @return ModuleOverview which contains the module and the list of sequences and its slides
+     */   
+    public SequenceOverview createSequenceOverviewNode(ISequence sequence) {
```

Comment:
this doesn't seem to be used anywhere but in this class, so it shouldn't be public (unless it's indented to be used anywhere else? in that case it should be part of the interface).